src/SFML/Graphics/Text.cpp

```diff
@@ -347,6 +347,7 @@ Vector2f Text::findCharacterPos(std::size_t index) const
         position.x += m_font->getGlyph(curChar, m_characterSize, isBold).advance + letterSpacing;
     }
 
+    position += m_bounds.getPosition();
```

Comment:
Not sure at all, but should this be

```suggestion
    position += getLocalBounds().getPosition();
```

so that `ensureGeometryUpdate()` get called beforehand if necessary? That function seems to be the one responsible for recalculating the bounds in case something changes in the text.

Reply:
BTW, it would also be nice to have a comment above this line explaining why the local bounds' position need to be added to the character position.

Reply:
ah I see. that's a good catch. Thanks! I didn't know we recalculated each time we call global bounds!